pkg/log/log.go

```diff
@@ -2,177 +2,162 @@ package log
 
 import (
 	"bytes"
-	"flag"
 	"fmt"
 	"io"
 	"net/http"
-	"os"
+	"strings"
+	"sync"
 	"time"
 
-	"github.com/operator-framework/operator-sdk/pkg/log/zap"
+	"github.com/codeready-toolchain/registration-service/pkg/middleware"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 
 	"github.com/gin-gonic/gin"
 	"github.com/go-logr/logr"
-	"github.com/spf13/pflag"
 )
 
 var (
-	log Logger
+	logger *Logger
+	once   sync.Once
 )
 
-// Log interface for the logger.
-type Log interface {
-	Errorf(ctx *gin.Context, err error, msg string, args ...string)
-	Infof(ctx *gin.Context, msg string, args ...string)
-	WithValues(keysAndValues ...interface{})
-	SetOutput(out io.Writer, isTestingMode bool)
+// Logger implements log.Logger
+type Logger struct {
+	logr logr.Logger
+	name string
 }
 
-type Logger struct {
-	lgr           logr.Logger
-	name          string
-	tags          []interface{}
-	out           io.Writer
-	isTestingMode bool
-}
-
-// InitializeLogger initializes the logger.
-func InitializeLogger(withName string) *Logger {
-
-	zapFlagSet := pflag.NewFlagSet("zap", pflag.ExitOnError)
```

Comment:
Not sure why we would need all these zap initialization. So I've removed it. It's not good what we get by the underling logger by default? Maybe I'm missing something.

Reply:
hmm i think it may be needed, it's used in our operators as well: https://github.com/codeready-toolchain/member-operator/blob/master/cmd/manager/main.go#L49

Reply:
Will need to take a closer look. But let’s bring it back for now then.

Reply:
I just double checked. We need these settings. Otherwise the logger is not initialized properly.